src/mem/cache/base.hh

```diff
@@ -1054,6 +1061,20 @@ class BaseCache : public ClockedObject, CacheAccessor
 
     int squashedWays;
 
+    std::vector<Tick> lastWriteFinishTick;
+
+    int getBank(Addr addr) {
+        return (addr >> blkAddrShift) & 0b11;
+    }
+
+    void writeBlock(Addr addr, Tick when) {
+        lastWriteFinishTick[getBank(addr)] = when + cyclesToTicks(dataLatency + Cycles(1));
+    }
+
+    bool isWriteBlock(Addr addr) {
+        return lastWriteFinishTick[getBank(addr)] > curTick();
+    }
```

Comment:
RTL's SRAM only blocks all read requests during the single writing cycle. It seems that `isWriteBlock` blocks all requests from the time the write request is received to the completion of writing?